mne/channels/tests/test_interpolation.py

```diff
@@ -76,8 +75,26 @@ def test_interplation():
     evoked.interpolate_bads_eeg()
     assert_array_equal(ave_after, evoked.data[bads_idx])
 
+    # make sure error is raised when preload is False
     for inst in [raw, epochs]:
         assert hasattr(inst, 'preload')
         inst.preload = False
         inst.info['bads'] = [inst.ch_names[1]]
         assert_raises(ValueError, inst.interpolate_bads_eeg)
+
+    # test interpolating different channels per epoch
+    epochsi = epochs_orig.copy()
+    epochs_12 = epochs_orig.copy()
+    epochs_12.info['bads'] = ['EEG 012']
+    epochs_12.interpolate_bads_eeg()
+    epochs_12_17 = epochs_orig.copy()
+    epochs_12_17.info['bads'] = ['EEG 012', 'EEG 017']
+    epochs_12_17.interpolate_bads_eeg()
+    _interpolate_bads_eeg_epochs(epochsi, [['EEG 012'], [],
+                                           ['EEG 012', 'EEG 017'], ['EEG 012'],
+                                           [], [], []])
+    assert_allclose(epochsi._data[0], epochs_12._data[0])
+    assert_array_equal(epochsi._data[1], epochs_orig._data[1])
+    assert_allclose(epochsi._data[2], epochs_12_17._data[2])
+    assert_allclose(epochsi._data[3], epochs_12._data[3])
```

Comment:
It's possible that the order of operations could differ slightly depending on the striding / array-continuous direction. You could try re-casting both to have the same striding to see if they match. I could also imagine that having continuity in one direction that allowed you to use SSE/AVX SIMD instructions could change the numbers slightly since the summation could effectively be binned differently. Can't remember how strict the linear algebra / implementations are for those operations...